AirCasting/SessionViews/EditViewModel.swift

```diff
@@ -0,0 +1,57 @@
+// Created by Lunar on 15/12/2021.
+//
+
+import Foundation
+import SwiftUI
+import CoreData
+
+class EditSessionViewModel: ObservableObject {
+    
+    @Published var sessionName = ""
+    @Published var sessionTags = ""
+    private let measurementStreamStorage: MeasurementStreamStorage
+    private let sessionUpdateService: SessionUpdateService
+
+    init(measurementStreamStorage: MeasurementStreamStorage, sessionUpdateService: SessionUpdateService) {
+        self.measurementStreamStorage = measurementStreamStorage
+        self.sessionUpdateService = sessionUpdateService
+    }
+    
+    func saveChanges(for uuid: SessionUUID, completion: @escaping () -> Void) {
+        guard !sessionName.isEmpty else { return }
+        let name = sessionName
+        let tags = sessionTags
+        measurementStreamStorage.accessStorage { [self] storage in
```

Comment:
It's difficult! I'm not good with those either. 
But as far as I understand these retain cycles, I think that both `[self]` and `[weak self]` are ok here. The only thing that will be holding EditViewModel would be storage, and it'll "let it go" (I have to learn more professional language in such cases :D ) after storage's block of this code is finished. So we shouldn't end up with a cycle. 
But like I said - I'm far from being expert -  maybe Ania or Paweł will have a strong opinion on this matter

Reply:
indeed @smialko , the block is not retained inside the storage, it will release as soon as it completes so no `[weak self]` is needed here since no retain cycle is really present.

Reply:
I will leave it and accept the [self] solution 😄